diff --git a/libraries/liblmdb/mdb.c b/libraries/liblmdb/mdb.c
index 8a62eff..1976340 100644
--- a/libraries/liblmdb/mdb.c
+++ b/libraries/liblmdb/mdb.c
@@ -113,6 +113,9 @@ typedef SSIZE_T     ssize_t;
 /* Most platforms have posix_memalign, older may only have memalign */
 #define HAVE_MEMALIGN  1
 #include <malloc.h>
+#if defined (__sun)
+#define _POSIX_PTHREAD_SEMANTICS       1
+#endif
 #endif

 #if !(defined(BYTE_ORDER) || defined(__BYTE_ORDER))
