--- mattermost/Makefile	Mon Aug  7 02:29:44 2017
+++ mattermost/Makefile	Mon Aug  7 02:36:08 2017
@@ -1,4 +1,4 @@
-.PHONY: build package run stop run-client run-server stop-client stop-server restart restart-server restart-client start-docker clean-dist clean nuke check-style check-client-style check-server-style check-unit-tests test dist setup-mac prepare-enteprise run-client-tests setup-run-client-tests cleanup-run-client-tests test-client build-linux build-osx build-windows internal-test-web-client vet run-server-for-web-client-tests
+.PHONY: build package run stop run-client run-server stop-client stop-server restart restart-server restart-client start-docker clean-dist clean nuke check-style check-client-style check-server-style check-unit-tests test dist setup-mac prepare-enteprise run-client-tests setup-run-client-tests cleanup-run-client-tests test-client build-solaris build-linux build-osx build-windows internal-test-web-client vet run-server-for-web-client-tests
 
 # For golang 1.5.x compatibility (remove when we don't want to support it anymore)
 export GO15VENDOREXPERIMENT=1
@@ -319,6 +319,10 @@
 	ln -s $(BUILD_ENTERPRISE_DIR) enterprise
 endif
 
+build-solaris: .prebuild prepare-enterprise
+	@echo Build Solaris amd64
+	env GOOS=solaris GOARCH=amd64 $(GO) install $(GOFLAGS) $(GO_LINKER_FLAGS) ./cmd/platform
+
 build-linux: .prebuild prepare-enterprise
 	@echo Build Linux amd64
 	env GOOS=linux GOARCH=amd64 $(GO) install $(GOFLAGS) $(GO_LINKER_FLAGS) ./cmd/platform
